packages/orchestration/src/orchestration-chat-completion-stream-response.ts

```diff
@@ -0,0 +1,57 @@
+import type { TokenUsage } from './client/api/schema/index.js';
+import type { OrchestrationChatCompletionStream } from './orchestration-chat-completion-stream.js';
+
+/**
+ * Orchestration chat completion stream response.
+ */
+export class OrchestrationChatCompletionStreamResponse<T> {
+  private _usage: TokenUsage | undefined;
+  /**
+   * Finish reasons for all choices.
+   */
+  private _finishReasons: Map<number, string> = new Map();
+  private _stream: OrchestrationChatCompletionStream<T> | undefined;
+
+  public getTokenUsage(): TokenUsage | undefined {
+    return this._usage;
+  }
+
+  /**
+   * @internal
+   */
+  _setTokenUsage(usage: TokenUsage): void {
+    this._usage = usage;
+  }
+
+  public getFinishReason(choiceIndex = 0): string | undefined | null {
```

Comment:
[req] I didn't see an example where `finish_reason` is `null`. I think if it is present it always has atleast an empty string.

```suggestion
  public getFinishReason(choiceIndex = 0): string | undefined {
```